src/Controller/UserController.php

```diff
@@ -12,7 +12,22 @@ class UserController extends Controller
      */
     public function login(): void
     {
-        $this->render('layout/login.html.twig');
+        $messages = [];
+        if ('POST' === $_SERVER['REQUEST_METHOD'] && !empty($_POST)) {
+            try {
+                $userId = $this->auth->authenticateUser($_POST);
```

Comment:
Set the session directly on authenticate user, the you have only one lign ` $this->auth->authenticateUser($_POST);` and not three